libenv/unix_iface.c

```diff
@@ -752,4 +755,389 @@ static int aix_get_mac_addr(const char *device_name, uint8_t mac[6])
+        JsonObjectRemoveKey(element, raw_key);
+        BufferDestroy(addr);
+    }
+}
+
+static long JsonExtractParsedNumber(JsonElement* element, const char* raw_key, const char *new_key, const bool hex_mode, const bool keep_number)
+{
+    long num = 0;
+
+    if (1 == sscanf(JsonObjectGetAsString(element, raw_key), hex_mode ? "%lx" : "%ld", &num))
+    {
+        if (!keep_number)
+        {
+            JsonObjectRemoveKey(element, raw_key);
+        }
+
+        if (NULL != new_key)
+        {
+            JsonObjectAppendInteger(element, new_key, num);
+        }
+    }
+
+    return num;
+}
+
+/*******************************************************************/
+
+static ProcPostProcessFn NetworkingRoutesPostProcessInfo(void *passed_ctx, void *json)
+{
+    EvalContext *ctx = passed_ctx;
+    JsonElement *route = json;
+
+    JsonRewriteParsedIPAddress(route, "raw_dest", "dest", false);
+    JsonRewriteParsedIPAddress(route, "raw_gw", "gateway", false);
+    JsonRewriteParsedIPAddress(route, "raw_mask", "mask", false);
+
+    // TODO: check that the metric and the others are decimal (ipv6_route uses hex for metric and others maybe)
+    JsonExtractParsedNumber(route, "metric", "metric", false, false);
+    JsonExtractParsedNumber(route, "mtu", "mtu", false, false);
+    JsonExtractParsedNumber(route, "refcnt", "refcnt", false, false);
+    JsonExtractParsedNumber(route, "use", "use", false, false);
+    JsonExtractParsedNumber(route, "window", "window", false, false);
+    JsonExtractParsedNumber(route, "irtt", "irtt", false, false);
+
+    JsonElement *decoded_flags = JsonArrayCreate(3);
+    long num_flags = JsonExtractParsedNumber(route, "raw_flags", NULL, true, false);
+
+    bool is_up = (num_flags & RTF_UP);
+    bool is_gw = (num_flags & RTF_GATEWAY);
+    bool is_host = (num_flags & RTF_HOST);
+    bool is_default_route = (0 == strcmp(JsonObjectGetAsString(route, "dest"), "0.0.0.0"));
+
+    const char* gw_type = is_gw ? "gateway":"local";
+
+    // These flags are always included on Linux in platform.h
+    JsonArrayAppendString(decoded_flags, is_up ? "up":"down");
+    JsonArrayAppendString(decoded_flags, is_host ? "host":"net");
+    JsonArrayAppendString(decoded_flags, gw_type);
+    JsonObjectAppendElement(route, "flags", decoded_flags);
+
+    if (is_up && is_gw)
+    {
+        Buffer *formatter = BufferNew();
+        BufferPrintf(formatter, "ipv4_gw_%s", JsonObjectGetAsString(route, "gateway"));
+        EvalContextClassPutHard(ctx, BufferData(formatter), "inventory,networking,/proc,source=agent,procfs");
+        BufferDestroy(formatter);
+
+        if (is_default_route)
+        {
+            EvalContextVariablePutSpecial(ctx,
+                                          SPECIAL_SCOPE_SYS, "default_gateway",
+                                          JsonObjectGetAsString(route, "gateway"), CF_DATA_TYPE_STRING,
+                                          "inventory,networking,/proc,source=agent,procfs");
```

Comment:
Nice, this negates the need for https://github.com/cfengine/masterfiles/pull/649